frontend/src/components/Banner.js

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2021 Christoph Ehm <christoph.ehmendoerfer@campus.tu-berlin.de>
+
+import React from "react";
+import HeaderTypography from "./HeaderTypography";
+import ParagraphTypography from "./ParagraphTypography";
+import { Link } from "react-router-dom";
+import RoundedEdgesButton from "./RoundedEdgesButton";
+
+import { textColor, activeTextColor } from "../assets/jss/colorPalette";
+import bannerImage from "../assets/img/banner.png";
+
+/**
+ * Displays a welcoming background image with a button.
+ * @param user user object of the currently logged in account
+ * @returns {JSX.Element}
+ */
+export default function Banner({ children, title, user }) {
+	const bannerTitle = <BannerTitle>{ title }</BannerTitle>;
+	const subTitle = <BannerSubTitle>{ children }</BannerSubTitle>;
+	const button = <BannerButton isLoggedIn={(user != null)} style={{ fontSize: "17px", }}/>;
+
+	const containerStyle = ({
+		height: "305px",
+	});
+	const backgroundImageStyle = ({
+		padding: "44px",
+		alignItems: "center",
+		justifyContent: "center",
+		textAlign: "center",
+	
+		borderRadius: "15px",
+		backgroundImage: `url(${bannerImage})`,
+		backgroundPosition: "center",
+		backgroundSize: "cover",
+	});
+
+	const BannerItem = ({ children, style }) => {
+		const itemStyle = ({
+			width: "100%",
+			display: "flex",
+			alignItems: "center",
+			justifyContent: "center",
+			...style,
+		});
+		return <div style={itemStyle}>{ children }</div>;
+	};
+
+	return (
+		<div style={containerStyle}>
+			<div style={backgroundImageStyle}>
+				<BannerItem style={{ marginBottom: "11px" }}>{ bannerTitle }</BannerItem>
+				<BannerItem style={{ height: "66px", marginBottom: "32px" }}>{ subTitle }</BannerItem>
+				<BannerItem style={{ }}>{ button }</BannerItem>
+			</div>
+		</div>
+	);
+}
+
+const newBannerStyle = (style) => ({ color: textColor, ...style, });
+
+function BannerTitle({ style, children }) {
+	const smallStyle = newBannerStyle({ fontSize: 30, fontWeight: 550, bottomMargin: "5px", ...style });
+	const mediumStyle = newBannerStyle({ fontSize: 37, fontWeight: 700, ...style });
+
+	return <HeaderTypography sx={smallStyle} md={mediumStyle}>{ children }</HeaderTypography>;
```

Comment:
Seems like `xs` and `md` are not working for this component?

Reply:
`sx` is the custom style property of MUI and `md` is not defined for HeaderTypography. React does not support `md` out of the box and for us only works for MUI components that know how to handle it. You can use something like [react-responsive](https://www.npmjs.com/package/react-responsive) to find out if the window width is above a certain threshold and based on that change the style.